weldx/transformations.py

```diff
@@ -1078,19 +1106,56 @@ def add_cs(
             An instance of
             weldx.transformations.LocalCoordinateSystem that describes how the new
             coordinate system is oriented in its parent system.
+        lsc_child_in_parent:
+            If set to 'True', the passed 'LocalCoordinateSystem' instance describes
+            the new system orientation towards is parent. If 'False', it describes
+            how the parent system is positioned in its new child system.
 
         """
         if not isinstance(local_coordinate_system, LocalCoordinateSystem):
             raise TypeError(
                 "'local_coordinate_system' must be an instance of "
                 + "weldx.transformations.LocalCoordinateSystem"
             )
-        self._check_coordinate_system_exists(reference_system_name)
 
-        self._add_coordinate_system_node(coordinate_system_name)
-        self._add_edges(
-            coordinate_system_name, reference_system_name, local_coordinate_system
-        )
+        if self.has_coordinate_system(coordinate_system_name):
+            # todo:
+            #  discuss: update and add functionality should be separated
+            #  why?   : to prevent errors. Misspelling of the system name might cause
+            #           unwanted updates or unwanted additions. Separate function can
+            #           catch that by knowing about the users intention.
```

Comment:
we should consider that

Reply:
having separate functions would complicate more "interactive" uses (change LCS but run same CSM code)

I think we can keep it like this for now, maybe add a nicer plot function to validate the CSM instance by the user instead :)